packages/components/mention/src/helper.ts

```diff
@@ -0,0 +1,204 @@
+
+export const filterOption = (
+  pattern: string,
+  option: MentionOption
+): boolean => {
+  const lowerCase = pattern.toLowerCase()
+  const label = option.label || option.value
+  return label.toLowerCase().includes(lowerCase)
+}
+
+export const getMentionCtx = (
+  inputEl: HTMLInputElement | HTMLTextAreaElement,
+  prefix: string | string[],
+  split: string
+) => {
+  const { selectionEnd } = inputEl
+  if (selectionEnd === null) return
+  const inputValue = inputEl.value
+  const prefixArray = typeof prefix === 'string' ? [prefix] : prefix
+  let splitIndex = -1
+  let mentionCtx: MentionCtx | undefined
+  for (let i = selectionEnd - 1; i >= 0; --i) {
+    const char = inputValue[i]
+    if (char === split || char === '\n' || char === '\r') {
+      splitIndex = i
+      continue
+    }
+    if (prefixArray.includes(char)) {
+      const end = splitIndex === -1 ? selectionEnd : splitIndex
+      const pattern = inputValue.slice(i + 1, end)
+      mentionCtx = {
+        pattern,
+        start: i + 1,
+        end,
+        prefix: char,
+        prefixIndex: i,
+        splitIndex,
+        selectionEnd,
+      }
+      break
+    }
+  }
+  return mentionCtx
+}
+
+/**
+ * fork from textarea-caret-position
+ * https://github.com/component/textarea-caret-position
+ */
+export const getCursorPosition = (
+  element: HTMLInputElement | HTMLTextAreaElement,
+  options = {
+    debug: false,
+    useSelectionEnd: false,
+  }
+) => {
+  const selectionStart =
+    element.selectionStart !== null ? element.selectionStart : 0
+  const selectionEnd = element.selectionEnd !== null ? element.selectionEnd : 0
+  const position = options.useSelectionEnd ? selectionEnd : selectionStart
+  // We'll copy the properties below into the mirror div.
+  // Note that some browsers, such as Firefox, do not concatenate properties
+  // into their shorthand (e.g. padding-top, padding-bottom etc. -> padding),
+  // so we have to list every single property explicitly.
+  const properties: string[] = [
+    'direction', // RTL support
+    'boxSizing',
+    'width', // on Chrome and IE, exclude the scrollbar, so the mirror div wraps exactly as the textarea does
+    'height',
+    'overflowX',
+    'overflowY', // copy the scrollbar for IE
+    'borderTopWidth',
+    'borderRightWidth',
+    'borderBottomWidth',
+    'borderLeftWidth',
+    'borderStyle',
+    'paddingTop',
+    'paddingRight',
+    'paddingBottom',
+    'paddingLeft',
+    // https://developer.mozilla.org/en-US/docs/Web/CSS/font
+    'fontStyle',
+    'fontVariant',
+    'fontWeight',
+    'fontStretch',
+    'fontSize',
+    'fontSizeAdjust',
+    'lineHeight',
+    'fontFamily',
+    'textAlign',
+    'textTransform',
+    'textIndent',
+    'textDecoration', // might not make a difference, but better be safe
+    'letterSpacing',
+    'wordSpacing',
+    'tabSize',
+    'MozTabSize',
+  ]
+  // Firefox 1.0+
+  const isFirefox = navigator.userAgent.toLowerCase().includes('firefox')
```

Comment:
Existing built-in methods
https://github.com/element-plus/element-plus/blob/cd3d58f07c823419665d232237e6e2eb012393ee/packages/utils/browser.ts#L3

Reply:
done